sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/handler/MessageWithDeliveryTag.java

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp.implementation.handler;
+
+import org.apache.qpid.proton.message.Message;
+import org.apache.qpid.proton.message.impl.MessageImpl;
+
+import java.util.UUID;
+
+/**
+ * Represents a received message containing a delivery tag.
+ */
+public final class MessageWithDeliveryTag extends MessageImpl {
```

Comment:
Ah yes. I think this originally existed in Service Bus? I don't know but I assume the original class is deleted now.

Reply:
Yes, this was uplifted to AMQP-core, so that upon receiving, we can decode immediately on the same reactor thread - That avoids thread hoping in SB, removes extra delivery map in core.

Nothing changed in our v1 code path - i.e. v1 Service Bus path continues to decode lazily and uses its local type to hold message and lock-token, and v1 amqp-core continues to manage that delivery map.

Reply:
Main goal is – to keep v1 code/logic as it is, and do the all deletions in one go when we decide to drop v1-v2 side by side support.